vunit/ui/__init__.py

```diff
@@ -752,6 +762,47 @@ def _main(self, post_run):
         all_ok = self._main_run(post_run)
         return all_ok
 
+    def _update_test_filter(self, dependency_files):
+        """
+        Update test filter to match testbenches depending on user_files
+        """
+        # Find source file objects corresponding to the provided files.
+        # Non-existing files, directories and files not added to the
+        # project are ignored.
+        dependency_paths = []
```

Comment:
I think it would be more efficient to use a `set()`. There will be no duplicates in `dependency_paths` even if the user passes duplicates, and the `if a in b` check below is much faster for set than list.